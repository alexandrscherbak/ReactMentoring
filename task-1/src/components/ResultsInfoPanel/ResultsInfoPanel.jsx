import React from 'react';
import ResultsCount from './ResultsCount/ResultsCount.jsx';
import ResultsSort from './ResultsSort/ResultsSort.jsx';

export default class ResultsInfoPanel extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div>
				<ResultsCount count={this.props.count}/>
				<ResultsSort/>

				<style jsx>
					{`
						div {
							display: flex;
							justify-content: space-between;
							background-color: #f5f5f5;
							color: #919191;
							font-weight: bold;
							height: 50px;
							line-height: 50px;
							padding: 0 100px;
						}
					`}
				</style> 
			</div>
		);
	}
};
