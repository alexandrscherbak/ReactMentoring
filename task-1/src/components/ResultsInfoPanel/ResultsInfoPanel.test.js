import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ResultsInfoPanel from './ResultsInfoPanel.jsx';
import renderer from 'react-test-renderer';

jest.mock('./ResultsCount/ResultsCount.jsx', () => () => 'ResultsCount');
jest.mock('./ResultsSort/ResultsSort.jsx', () => () => 'ResultsSort');
Enzyme.configure({ adapter: new Adapter() });

function setup() {
	return mount(<ResultsInfoPanel/>);
}

describe('ResultsInfoPanel', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<ResultsInfoPanel/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.length).toBe(1);
	});
});
