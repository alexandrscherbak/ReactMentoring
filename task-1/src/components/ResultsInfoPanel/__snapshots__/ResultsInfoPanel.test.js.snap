// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsInfoPanel snapshot 1`] = `
<div
  className="jsx-1938216586"
>
  ResultsCount
  ResultsSort
</div>
`;
